metadata/src/main/java/com/redhat/lightblue/metadata/mongo/MongoMetadata.java

```diff
@@ -223,6 +216,15 @@ public EntityInfo getEntityInfo(String entityName) {
         } catch (Error e) {
             // rethrow lightblue error
             throw e;
+
+        } catch (CommandFailureException e) {
+            // give a better exception in case auth failed
+            LOGGER.error(e.getMessage(), e);
+            if(e.getCode() == 18){
```

Comment:
It is from mongodb, https://github.com/lightblue-platform/lightblue-rest/issues/13 and we can also simulate it, but it is the MongoDB itself that generate this message for this case https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.err#L21